### ISSUE TITLE
Investigate playability issues with The Fourth Generation (1996)

### ISSUE THREAD

**Issue by kcgen** [dosbox-staging/dosbox-staging#2871]
### Context

Currently the eXoDOS project runs this game using [the DOSBox Daum fork](https://github.com/ykhwong/dosbox-svn-daum), which is understood to have specific emulation fixes not currently in upstream DOSBox (0.74.x and SVN) or active forks such as X, Pure, and Staging.

DOSBox Daum had 51 releases that included DOSBox SVN's commits ~r3600 through ~r3900, [spanning 2010 through 2015](http://ykhwong.x-y.net/).

### The goal of this ticket is to answer the following

Is Daum _still_ needed for this game?
- [ ] Does DOSBox SVN ([download latest](http://www.emucr.com/2023/09/dosbox-svn-r4483.html)) play it OK?
- [ ] Does DOSBox X ([download latest](https://dosbox-x.com/)) play it OK?
- [ ] Does DOSBox Staging ([download latest](https://dosbox-staging.github.io/downloads/development-builds/)) play it OK?

If Daum plays it better:
- [ ] What aspect is better?
- [ ] Which Daum release was the **_first_** to fix it?
    Use these builds [here](http://ykhwong.x-y.net/subpages/old.php) to narrow it down.

### How you can help

 1. Download the above lastest DOSBox SVN, X, and Staging builds.
 1. Ensure you have [eXoDOS v5](https://www.retro-exo.com/exodos.html) (or eventually v6)
 1. Find this game inside the eXoDOS installed folders (look for: `4thgen/`)
 1. Run the game using the DOSBox SVN, X, and Staging builds.
    _If it plays fine (everything looks, sounds, and handles OK) then simply indicate as such. Your task is done!_

If SVN, X, and Staging all have problems, then try it with Daum:
 1. Download the [first and last Daum releases](http://ykhwong.x-y.net/subpages/old.php).
    Note: _Daum had the most releases for Microsoft Windows, so having a Windows system (or VM) will help the most to answer the last question above._
 1. Does the latest Daum release play it correctly? If so: summarize/describe how it's better than DOSBox SVN, X, and Staging (screenshots, video, audio records all help too).
 1. Find the earliest (or oldest) Daum release that fixed it.
    1. Run the Daum release half way between the known good release (where the game playing correctly) and bad release (where the game doesn't playing correctly) until you've narrowed it down to adjacent releases.
    1. It will take _at-most_ seven runs of different Daum versions to narrow this down (using this middle-selecting technique); so it's not as daunting as it seems! :-)

---

### Follow up by code contributors

If Daum plays it better, your answers (especially the last one that narrows down the fix to a specific release) are critical to helping isolate the chunk of code fixed the given bug.

Pull requests should credit:
- @ykhwong as the original code author,
- the test team member who confirmed and isolated the Daum release,
- yourself (PR author) as integrating the fix

The PR should include the `imported-from:` URL pointing to DOSBox Daum. For an example, see commit a35c7c889.

**Comment by ebenbruyns** [comment]
I have tried this on staging on linux and it behaves like daum on windows (I ran it in wine so keep that in mind) and linux. Does anybody know if there are specific issues with this game?

**Comment by rderooy** [comment]
Tested on Linux with Wayland.

## DOSBox 0.74-3
Once again cursor keys don't work. Numeric keypad does work.
Apart from that seems to work.

## DOSBox Daum
Crashes on startup with:
```
E_Exit: DRC64:Unhandled memory reference

ROMBIOS memory dump:
     0x000e0000-0x000f06ff free=1 
     0x000f0700-0x000f070d free=0 DOSBox vm86 hack
     0x000f070e-0x000f0fff free=1 
     0x000f1000-0x000f25ff free=0 DOSBox callbacks region
     0x000f2600-0x000fdfef free=1 
     0x000fdff0-0x000fdfff free=0 BIOS configuration (INT 15h AH=0xC0)
     0x000fe000-0x000fffef free=0 BIOS with fixed layout
[end dump]
TIMER: 1 leftover handlers (clean up!).
Exit to error: DRC64:Unhandled memory reference
```

## DOSBox Staging 0.80.1
Black screen initially on startup until Enter is pressed, then the intro starts.
After that works fine. No issues with the cursor keys.

## DOSBox Staging Git
Same as 0.80.1

## DOSBox-X 2023.09.01
Works fine, intro starts straight away. No issues with the cursor keys.

# dosbox.conf
```ini
# Config file for Dosbox-SVN Daum Jan 25, 2015
[sdl]
fullscreen=false
fulldouble=false
fullresolution=0x0
windowresolution=1280x960
output=direct3d
mapperfile=mapper.map
usescancodes=true
[dosbox]
machine=svga_s3
memsize=16
[render]
aspect=true
scaler=normal2x
[cpu]
core=auto
cputype=auto
cycles=25000
[mixer]
[midi]
[sblaster]
[gus]
[speaker]
disney=true
[joystick]
[serial]
[dos]
[ipx]
[autoexec]
mount c .\eXoDOS\4thgen
imgmount d .\eXoDOS\4thgen\CD\TFG01.iso -t cdrom
c:
@cd 4GEN
@cls
@4gen
exit
```

**Comment by johnnovak** [comment]
It would be good to create another ticket for the difference with X perhaphs @rderooy .

**Comment by Burrito78** [comment]
Results (macOS Ventura 13.5.2/Apple M2):

**DOSBox Staging 0.81.0-alpha-1689**
All OK

**DOSBox SVN	snapshot 12.09.23**
All OK

**DOSBox-X v2023.09.01**
All OK

I don't run Windows so i can't test with Daum.

**Comment by Burrito78** [comment]
@rderooy Robert, do you think we can close this one?

**Comment by kcgen** [comment]
Yes, the game should start up and present initial graphics without needing extra keystrokes to 'coax' it along. 

(Not at my desk), but do we know if SVN also starts it properly similar to 0.74 (without needing keystrokes)?

**Comment by kcgen** [comment]
Confirmed DOSBox SVN requires an `Enter` tap to progress to the intro (here it is, waiting):

![2023-09-25_11-39](https://github.com/dosbox-staging/dosbox-staging/assets/1557255/699f1fc6-f488-4b3b-a100-3c0fa5a2e7aa)

**Comment by rderooy** [comment]
As mentioned, this happens with Staging, but not with DOSBox-X.

But as I was intending to setup an external video recording of this issue, I added a `pause` command before starting the game. But then I discovered that adding the `pause` just before "fixed" the problem. Obviously you need to hit a key to continue due to the pause, but that simultaneously fixes the black screen pause issue.

**Comment by kcgen** [comment]
@rderooy , @Burrito78 - can you double check this "extra keystroke" needed when running 0.74.x?

I am seeing the same behavior as SVN.. 0.74.x similarly needs a keystroke to continue on and bring up the intro:

### 0.74.3

![2023-09-25_13-44](https://github.com/dosbox-staging/dosbox-staging/assets/1557255/a647d4ba-72c9-49fe-b68a-5c9460576f9b)

### 0.74.2

![2023-09-25_13-50](https://github.com/dosbox-staging/dosbox-staging/assets/1557255/b20703b8-04aa-474b-81fc-1a7f53dd33cf)

**Comment by kcgen** [comment]
ECE, for me, also needs the extra keystroke:

![2023-09-25_13-53](https://github.com/dosbox-staging/dosbox-staging/assets/1557255/eed45937-a761-4bed-b5a3-59cc0df61771)

**Comment by rderooy** [comment]
- DOSBox ECE r4230, yes
- DOSBox 0.74-3, no

**but** for ECE I am running the eXeDOS version via Wine, while for DOSBox 0.74-3 I am running the Linux flatpak build (it is a pain to compile the original DOSBox these days on a modern distro).

I did not see a regular DOSBox 0.74 Windows build in the eXeDOS tree. They seem to all be different ECE or Daum variants.

**Comment by rderooy** [comment]
Never mind, did not look properly. There is a 074r3 folder.

With the Windows version, of DOSBox 0.74-3 I get the black screen.

**Comment by kcgen** [comment]
Phew! thanks, @rderooy !

**Comment by Python-Exoproject** [comment]
I ran this game with current 81.0 release and found no issues, happy for this to be closed from the exodos side if anyone else confirms

**Comment by johnnovak** [comment]
Thanks for confirming @Python-Exoproject
